tensorflow_federated/python/learning/keras_utils.py

```diff
@@ -111,14 +123,49 @@ def from_keras_model(keras_model,
     ValueError: If `keras_model` was compiled.
   """
   py_typecheck.check_type(keras_model, tf.keras.Model)
+  py_typecheck.check_type(loss, (tf.keras.losses.Loss,
+                                 collections.Sequence,
+                                 collections.Mapping))
+
+  if loss_weights is not None:
+    py_typecheck.check_type(loss, (collections.Sequence,
+                                   collections.Mapping))
+
+  if isinstance(loss, collections.Sequence):
+    if len(loss) != len(keras_model.outputs):
+      raise ValueError('`keras_model` must have equal number of '
+                       'outputs and losses')
```

Comment:
TFF generally aims to include the parameters in the error string for easier debugging, this is probably different than traditional Python.

In this case, perhaps we could add the string `'loss: {}\noutputs: {}'.format(loss, keras_model.outputs)`. Same fro the error raised below.